File: 151. Reverse Words in a String/151. Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {
        // Trim leading and trailing spaces
        int left = 0, right = s.length() - 1;
        while (left <= right && s[left] == ' ') left++;
        while (right >= left && s[right] == ' ') right--;
        
        vector<string> words;
        string word;
        
        // Extract words
        for (int i = left; i <= right; i++) {
            if (s[i] != ' ') {
                word += s[i];
            } else if (!word.empty()) {
                words.push_back(word);
                word.clear();
            }
        }
        if (!word.empty()) {
            words.push_back(word);
        }
        
        // Reverse and join words
        reverse(words.begin(), words.end());
        string result;
        for (const auto& w : words) {
            if (!result.empty()) result += " ";
            result += w;
        }
        
        return result;
    }
};

File: 151. Reverse Words in a String/151. Reverse Words in a String.java
class Solution {
    public String reverseWords(String s) {
        // Trim leading and trailing spaces
        s = s.trim();
        
        // Split the string into words
        String[] words = s.split("\\s+");
        
        // Reverse the array of words
        for (int i = 0; i < words.length / 2; i++) {
            String temp = words[i];
            words[i] = words[words.length - 1 - i];
            words[words.length - 1 - i] = temp;
        }
        
        // Join the words with a single space
        return String.join(" ", words);
    }
}